### PR TITLE
Processing CSV

### PR DESCRIPTION
**Pull request by tanmayranjan**

### CODE REVIEW

**Review by maheshkumargangula — APPROVED** [review]

File: mvc-api/mvc-service/app/controllers/MVCContentController.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+class MVCContentController @Inject()(cc: ControllerComponents) (implicit exec: ExecutionContext) extends SearchBaseController(cc) {
+  val readJson: ReadJson = new ReadJson()
+  val readcsv: CSVContentProcessor = new CSVContentProcessor()
+  def createUsingCSV() = Action.async { implicit request =>
+    val body = request.body.asJson.getOrElse("{}").toString
+    var result = ResponseHandler.OK()
+    @transient val mapper = new ObjectMapper();
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/controllers/MVCContentController.scala]: This should not be inside a method. It is used multiple times.

File: mvc-api/mvc-service/app/controllers/MVCContentController.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      result = ResponseHandler.ERROR(ResponseCode.CLIENT_ERROR,"400","Bad Request,Please provide valid input")
+    }
+    val response = mapper.writeValueAsString(result);
+    Future(Ok(response).as("application/json"))
+  }
+  def createUsingJson() = Action.async { implicit request =>
+    val body = request.body.asJson.getOrElse("{}").toString
+    var result = ResponseHandler.OK()
+    @transient val mapper = new ObjectMapper();
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/controllers/MVCContentController.scala]: Same as above comment.

File: mvc-api/mvc-service/app/managers/CSVContentProcessor.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.sunbird.search.util.SearchConstants;
+
+import java.io.*;
+import java.util.Arrays;
+
+public class CSVContentProcessor {
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/managers/CSVContentProcessor.java:12]: Please use any CSV parsing opensource library to read and parse the data.
Also, I don't see any limit on # of lines in the file.

File: mvc-api/mvc-service/app/managers/EventObjectProducer.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.json.simple.parser.ParseException;
+import org.sunbird.search.util.SearchConstants;
+
+import java.util.UUID;
+
+public class EventObjectProducer {
+    EventProducer eventProducer = new EventProducer();
+    public void addToEventObj(JSONObject content, String contentId) throws ParseException {
+        // Add label
+        content.put("label","MVC");
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/managers/EventObjectProducer.java:16]: Please don't use `label` as metadata to identify an MVC content. This name is too generic.

File: mvc-api/mvc-service/app/managers/CSVContentProcessor.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                }
+            } catch (FileNotFoundException e) {
+
+            } catch (IOException e) {
+
+            }
+        }
+        catch (Exception e) {
+
+        }
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/managers/CSVContentProcessor.java]: Exceptions are suppressed. If the data is not valid then it won't process those records.

File: mvc-api/mvc-service/app/managers/Postman.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+import org.apache.http.message.BasicHeader;
+import org.apache.http.protocol.HTTP;
+import org.json.JSONObject;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class Postman {
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/managers/Postman.java:22]: remove the commented code.

File: mvc-api/mvc-service/app/managers/ReadJson.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
+package managers;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+public class ReadJson {
+    GetContentDefinition getContentDefinition = new GetContentDefinition();
+    public void read(String json) {
+
+        JSONParser parser = new JSONParser();
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/app/managers/ReadJson.java]: We always use ObjectMapper from to parse JSON string. Please don't add new libraries.

File: mvc-api/mvc-service/conf/routes [hunk 8]
```diff
@@ -6,6 +6,8 @@ GET		/health						controllers.HealthController.health()
 #POST    /v2/search						controllers.SearchController.search()
 #POST    /v3/search						controllers.SearchController.search()
 POST    /v3/mvcsearch                   controllers.SearchController.search()
+POST    /v3/mvccontent/create/json      controllers.MVCContentController.createUsingJson()
+POST    /v3/mvccontent/create/csv       controllers.MVCContentController.createUsingCSV()
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/conf/routes]: Change the Path to `/v3/mvccontent/import`

File: mvc-api/mvc-service/conf/routes [hunk 9]
```diff
@@ -6,6 +6,8 @@ GET		/health						controllers.HealthController.health()
 #POST    /v2/search						controllers.SearchController.search()
 #POST    /v3/search						controllers.SearchController.search()
 POST    /v3/mvcsearch                   controllers.SearchController.search()
+POST    /v3/mvccontent/create/json      controllers.MVCContentController.createUsingJson()
```
Comment (maheshkumargangula) [inline mvc-api/mvc-service/conf/routes]: Change the Path to `/v3/mvccontent/create`.

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: mvc-api/mvc-service/app/controllers/MVCContentController.scala (hunk 1 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/app/controllers/MVCContentController.scala]: okay will make this change

File: mvc-api/mvc-service/app/controllers/MVCContentController.scala (hunk 2 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/app/controllers/MVCContentController.scala]: okay will make this change

File: mvc-api/mvc-service/app/managers/CSVContentProcessor.java (hunk 3 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/app/managers/CSVContentProcessor.java:12]: But i have used BufferedReader which is used to process data with large csv files.
should limit on number of rows be implemented?

File: mvc-api/mvc-service/app/managers/EventObjectProducer.java (hunk 4 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/app/managers/EventObjectProducer.java:16]: label is the sole separator decided and we are using this everywhere
please discuss some parameter with sumit if this needs a change

File: mvc-api/mvc-service/app/managers/CSVContentProcessor.java (hunk 5 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/app/managers/CSVContentProcessor.java]: will make the change

File: mvc-api/mvc-service/app/managers/Postman.java (hunk 6 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/app/managers/Postman.java:22]: okay

File: mvc-api/mvc-service/conf/routes (hunk 8 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/conf/routes]: okay

File: mvc-api/mvc-service/conf/routes (hunk 9 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/conf/routes]: okay

File: mvc-api/mvc-service/app/managers/ReadJson.java (hunk 7 above)
Comment (tanmayranjan) [reply mvc-api/mvc-service/app/managers/ReadJson.java]: okay will make this change and remove the added library

**Pull request closed without merging**